components/core/src/clp/hash_utils.cpp

```diff
@@ -0,0 +1,107 @@
+#include "hash_utils.hpp"
+
+#include <span>
+#include <string>
+#include <vector>
+
+#include <fmt/format.h>
+#include <openssl/hmac.h>
+#include <openssl/sha.h>
+
+#include "ErrorCode.hpp"
+#include "spdlog_with_specializations.hpp"
+
+using std::span;
+using std::string;
+using std::vector;
+
+namespace clp {
+auto EvpDigestContext::digest_update(std::span<unsigned char const> input) -> ErrorCode {
+    if (m_is_digest_finalized) {
+        return ErrorCode_Unsupported;
+    }
+    if (1 != EVP_DigestUpdate(m_md_ctx, input.data(), input.size())) {
+        return ErrorCode_Failure;
+    }
+    return ErrorCode_Success;
+}
+
+auto EvpDigestContext::digest_final_ex(std::vector<unsigned char>& hash) -> ErrorCode {
+    if (m_is_digest_finalized) {
+        return ErrorCode_Unsupported;
+    }
+
+    hash.resize(SHA256_DIGEST_LENGTH);
```

Comment:
```suggestion
    hash.resize(EVP_MD_CTX_size(m_md_ctx));
```

Since the context may be initialized with any type, not just SHA256.